src/database.cpp

```diff
@@ -567,3 +605,52 @@ void silo::Database::load(const std::string& save_dir) {
       ia >> partitions[i];
    });
 }
+void silo::Database::preprocessing(
+   const PreprocessingConfig& config) {
+   std::cout << "build_pango_defs" << std::endl;
+   std::ifstream metadata_stream(config.metadata_file.relative_path());
+   pango_def = std::make_unique<pango_descriptor_t>(silo::build_pango_defs(alias_key, metadata_stream));
+
+   std::cout << "build_partitioning_descriptor" << std::endl;
+   part_def = std::make_unique<partitioning_descriptor_t>(silo::build_partitioning_descriptor(*pango_def, architecture_type::max_partitions));
+
+   std::cout << "partition_sequences" << std::endl;
+   std::ifstream metadata_stream2(config.metadata_file.relative_path());
+   istream_wrapper sequence_stream(config.sequence_file.relative_path());
+   partition_sequences(
+      *part_def,
+      metadata_stream2,
+      sequence_stream.get_is(),
+      config.partition_folder.relative_path(),
+      alias_key,
+      config.metadata_file.extension(),
+      config.sequence_file.extension());
+
+   std::cout << "sort_chunks" << std::endl;
+   silo::sort_chunks(
+      *part_def,
+      config.partition_folder.relative_path(),
+      config.metadata_file.extension(),
+      config.sequence_file.extension());
+
+   std::cout << "Dictionary" << std::endl;
+   dict = std::make_unique<Dictionary>();
+   for (size_t i = 0; i < part_def->partitions.size(); ++i) {
+      const auto& part = part_def->partitions.at(i);
+      for (unsigned j = 0; j < part.chunks.size(); ++j) {
+         std::string name = std::string(config.partition_folder.relative_path()) + chunk_string(i, j) + std::string(config.metadata_file.extension());
+         std::ifstream meta_in(name);
+         if (!meta_in) {
+            std::cerr << "Meta_data file " << name << " not found." << std::endl;
```

Comment:
```suggestion
            throw silo::PreprocessingException("Metadata partition file " + name + " not found.");
```
I think a dedicated `PreprocessingException` makes sense to have more specialized exceptions.